arbos/l2pricing/model.go

```diff
@@ -21,18 +22,43 @@ const InitialPricingInertia = 102
 const InitialBacklogTolerance = 10
 const InitialPerTxGasLimitV50 uint64 = 32 * 1000000
 
+const ConstraintDivisorMultiplier = 30
+
 func (ps *L2PricingState) AddToGasPool(gas int64) error {
 	backlog, err := ps.GasBacklog()
 	if err != nil {
 		return err
 	}
-	// pay off some of the backlog with the added gas, stopping at 0
+	backlog = addToBacklog(backlog, gas)
+	return ps.SetGasBacklog(backlog)
+}
+
+func (ps *L2PricingState) AddToGasPoolMultiConstraints(gas int64) error {
+	constraintsLength, err := ps.constraints.Length()
+	if err != nil {
+		return fmt.Errorf("failed to get number of constraints: %w", err)
+	}
+	for i := range constraintsLength {
+		constraint := ps.OpenConstraintAt(i)
+		backlog, err := constraint.backlog.Get()
+		if err != nil {
+			return fmt.Errorf("failed to get backlog of constraint %v: %w", i, err)
+		}
+		err = constraint.backlog.Set(addToBacklog(backlog, gas))
+		if err != nil {
+			return fmt.Errorf("failed to set backlog of constraint %v: %w", i, err)
+		}
+	}
+	return nil
+}
+
+// addToBacklog grows the backlog if the gas is negative and pays off  if the gas is positive.
+func addToBacklog(backlog uint64, gas int64) uint64 {
```

Comment:
I suggest to rename it to `applyGasDelta`, since it used both for backlog growth and decay